clients/auth/auth.go

```diff
@@ -0,0 +1,431 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/clients"
+	"github.com/influxdata/influx-cli/v2/pkg/influxid"
+)
+
+type Client struct {
+	clients.CLI
+	api.AuthorizationsApi
+	api.UsersApi
+}
+
+const (
+	ReadAction  = "read"
+	WriteAction = "write"
+)
+
+type token struct {
+	ID          string   `json:"id"`
+	Description string   `json:"description"`
+	Token       string   `json:"token"`
+	Status      string   `json:"status"`
+	UserName    string   `json:"userName"`
+	UserID      string   `json:"userID"`
+	Permissions []string `json:"permissions"`
+}
+
+type printParams struct {
+	deleted bool
+	token   *token
+	tokens  []token
+}
+
+type CreateParams struct {
+	clients.OrgParams
+	User        string
+	Description string
+
+	WriteUserPermission bool
+	ReadUserPermission  bool
+
+	WriteBucketsPermission bool
+	ReadBucketsPermission  bool
+
+	WriteBucketPermissions []string
+	ReadBucketPermissions  []string
+
+	WriteTasksPermission bool
+	ReadTasksPermission  bool
+
+	WriteTelegrafsPermission bool
+	ReadTelegrafsPermission  bool
+
+	WriteOrganizationsPermission bool
+	ReadOrganizationsPermission  bool
+
+	WriteDashboardsPermission bool
+	ReadDashboardsPermission  bool
+
+	WriteCheckPermission bool
+	ReadCheckPermission  bool
+
+	WriteNotificationRulePermission bool
+	ReadNotificationRulePermission  bool
+
+	WriteNotificationEndpointPermission bool
+	ReadNotificationEndpointPermission  bool
+
+	WriteDBRPPermission bool
+	ReadDBRPPermission  bool
+}
+
+func (c Client) Create(ctx context.Context, params *CreateParams) error {
+
+	orgID := params.OrgID.String()
+	orgName := params.OrgName
+	bucketPerms := []struct {
+		action string
+		perms  []string
+	}{
+		{action: ReadAction, perms: params.ReadBucketPermissions},
+		{action: WriteAction, perms: params.WriteBucketPermissions},
+	}
+
+	var permissions []api.Permission
+	for _, bp := range bucketPerms {
+		for _, p := range bp.perms {
+			// verify the input ID
+			if _, err := influxid.IDFromString(p); err != nil {
+				return fmt.Errorf("invalid bucket ID '%s': %w (did you pass a bucket name instead of an ID?)", p, err)
+			}
+
+			newPerm := api.Permission{
+				Action:   bp.action,
+				Resource: makePermResource("buckets", p, orgID, orgName),
+			}
+			permissions = append(permissions, newPerm)
+		}
+	}
+
+	providedPerm := []struct {
+		readPerm, writePerm bool
+		ResourceType        string
```

Comment:
Nitpick: Any reason to switch `lowerCase` and `UpperCase` in the field names here?